lms/views/lti/deep_linking.py

```diff
@@ -137,6 +132,34 @@ def file_picker_to_form_fields_v13(self):
             )
         }
 
+    @view_config(route_name="lti.deep_linking.form_fields.v11")
+    def file_picker_to_form_fields_v11(self):
+        """
+        Return a JSON-LD `ContentItem` representation of the LTI content.
+
+        See https://www.imsglobal.org/specs/lticiv1p0/specification.
+        """
+        url = self._content_to_url(
+            self.request.route_url("lti_launches"),
+            self.request.parsed_params["content"],
+            self.request.parsed_params.get("extra_params"),
+        )
```

Comment:
This is a repeated chunk from above. Maybe all this could move inside `_content_to_url` and this could be `_get_content_url(self.request)` or something?

I think this actually makes it easier to understand as where these values come from is right next to the code that uses them rather than separated as it is now:

```python
    @classmethod
    def _get_content_url(cls, request):
        """
        Translate content information from the frontend to a launch URL.

        We submit the content information to the LMS as an URL pointing to our
        `lti_launches` endpoint with any information required to identity
        the content as query parameters.
        """

        content = request.parsed_params["content"]
        params = dict(request.parsed_params.get("extra_params") or {})

        if content["type"] == "file":
            params["canvas_file"] = "true"
            params["file_id"] = content["file"]["id"]
        elif content["type"] == "url":
            params["url"] = content["url"]
        else:
            raise ValueError(f"Unknown content type: '{content['type']}'")

        return (
            urlparse(request.route_url("lti_launches"))
            ._replace(query=urlencode(params))
            .geturl()
        )
```